crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -30,84 +30,70 @@ impl ClusterOverviewChart {
-                )
-                .x_axis(
-                    Axis::default()
-                        .title("time ->")
-                        .style(Style::default().fg(Color::Gray))
-                        .bounds([
-                            get_time_as_secs(SystemTime::now()) as f64
-                                - self.view_size.as_secs() as f64,
-                            get_time_as_secs(SystemTime::now()) as f64,
-                        ]),
-                )
-                .y_axis(
-                    Axis::default()
-                        .style(Style::default().fg(Color::Gray))
-                        .bounds([0.0, max_workers_in_history as f64])
-                        .labels(vec![
-                            Span::from(0.to_string()),
-                            Span::from(max_workers_in_history.to_string()),
-                        ]),
-                );
-            frame.render_widget(chart, rect);
-        } else {
-            let header_text = vec![Spans::from("No workers connected")];
-            let text_block = Block::default()
-                .borders(Borders::ALL)
-                .style(Style::default().bg(Color::Black));
-            let paragraph = Paragraph::new(header_text)
-                .block(text_block)
-                .alignment(Alignment::Left)
-                .wrap(Wrap { trim: true });
-            frame.render_widget(paragraph, rect);
-        }
+        let chart = Chart::new(datasets)
+            .block(
+                Block::default()
+                    .title(Span::styled(
+                        "Worker Connection History",
+                        Style::default()
+                            .fg(Color::White)
+                            .add_modifier(Modifier::BOLD),
+                    ))
+                    .borders(Borders::ALL),
+            )
+            .x_axis(
+                Axis::default()
+                    .title("time ->")
+                    .style(Style::default().fg(Color::Gray))
+                    .bounds([
+                        (get_time_as_secs(SystemTime::now()) - self.view_size.as_secs()) as f64,
+                        get_time_as_secs(SystemTime::now()) as f64,
+                    ]),
+            )
+            .y_axis(
+                Axis::default()
+                    .style(Style::default().fg(Color::Gray))
+                    .bounds([0.0, max_workers_in_view])
+                    .labels(vec![
+                        Span::from(0.to_string()),
+                        Span::from(max_workers_in_view.to_string()),
+                    ]),
+            );
+        frame.render_widget(chart, rect);
     }
 
-    /// Returns the maximum number of workers connected in the past n seconds specified by view_size
-    fn get_max_workers_in_view(&self) -> Option<u32> {
+    /// Returns the worker connection chart data for the view size
+    fn get_workers_in_view(&self) -> Vec<(u64, u64)> {
```

Comment:
or should I rename it to get_data_in_view or something?

Reply:
It's ok like this.